docs/src/pages/components/data-grid/getting-started/migration-v4.md

```diff
@@ -0,0 +1,432 @@
+
+```jsx
+import { createMuiTheme as createThemeV4 } from '@material-ui/core/styles';
+import { createTheme as createThemeV5 } from '@mui/material/styles';
+
+const themeV4 = createThemeV4({
+  palette: {
+    primary: {
+      main: '#2196f3',
+    },
+  },
+});
+
+const themeV5 = createThemeV5({
+  palette: {
+    primary: {
+      main: theme.palette.primary.main,
+    },
+  },
+});
+```
+
+4. Apply the class name generator and v5 theme to the application.
+
+```jsx
+import * as React from 'react';
+import { ThemeProvider as ThemeProviderV5 } from '@mui/material/styles';
+import { ThemeProvider as ThemeProviderV4, StylesProvider } from '@material-ui/core/styles';
+
+const generateClassName = createGenerateClassName({ ... });
+const themeV4 = createThemeV4({ ... });
+const themeV5 = createThemeV5({ ... });
+
+export default function DataGridDemo() {
+  return (
+    <StylesProvider generateClassName={generateClassName}>
+      <ThemeProviderV4 theme={themeV4}>
+        <ThemeProviderV5 theme={themeV5}>
+          {/* Your component tree. */}
+        </ThemeProviderV5>
+      </ThemeProviderV4>
+    </StylesProvider>
+  );
+}
+```
+
+The following interactive demo shows how these steps tie together:
+
+{{"demo": "pages/components/data-grid/getting-started/DataGridV5WithCoreV4.js", "hideToolbar": true, "bg": true}}
+
+## Migrating MUI X from v4
+
+### CSS classes
+
+- Some CSS classes were removed
+
+  1. `.MuiDataGrid-window` was removed
+  2. `.MuiDataGrid-windowContainer` was removed
+
+- Some CSS classes were renamed
+
+  1. `.MuiDataGrid-viewport` was renamed `.MuiDataGrid-virtualScroller`
+  2. `.MuiDataGrid-dataContainer` was renamed `.MuiDataGrid-virtualScrollerContent`
+  3. `.MuiDataGrid-renderingZone` was renamed `.MuiDataGrid-virtualScrollerRenderZone`
+  4. `.MuiDataGrid-gridMenuList` was renamed `.MuiDataGrid-menuList`
+  5. `.MuiGridToolbarContainer-root` was renamed `.MuiDataGrid-toolbarContainer`
+  6. `.MuiGridMenu-root` was renamed `.MuiDataGrid-menu`
+  7. `.MuiDataGridColumnsPanel-root` was renamed `.MuiDataGrid-columnsPanel`
+  8. `.MuiGridPanel-root` was renamed `.MuiDataGrid-panel`
+  9. `.MuiGridPanelContent-root` was renamed `.MuiDataGrid-panelContent`
+  10. `.MuiGridPanelFooter-root` was renamed `.MuiDataGrid-panelFooter`
+  11. `.MuiDataGridPanelHeader-root` was renamed `.MuiDataGrid-panelHeader`
+  12. `.MuiGridPanelWrapper-root` was renamed `.MuiDataGrid-panelWrapper`
+  13. `.MuiGridFilterForm-root` was renamed `.MuiDataGrid-filterForm`
+  14. `.MuiGridToolbarFilterButton-root` was renamed `.MuiDataGrid-toolbarFilterList`
+
+### Module augmentation
+
+- The module augmentation is no longer enabled by default. This change was done to prevent conflicts with projects using `DataGrid` and `DataGridPro` together.
+
+  In order to still be able to do overrides at the theme level, add the following imports to your project:
```

Comment:
Probably a stupid question, but where should it be added?

Reply:
Anywhere I think.
It is just to let TypeScript knows the extended themes.
Usually we do it at the root of the project.

@m4theushw this is the wording from the CHANGELOG.
Maybe we should add precise `add the following imports to the entrypoint of your project`